packages/default-storage/ios/RNCAsyncStorage.mm

```diff
@@ -787,7 +787,8 @@ - (BOOL)_passthroughDelegate
             if (value) {
                 NSError *jsonError;
                 NSMutableDictionary *mergedVal = RCTJSONParseMutable(value, &jsonError);
-                if (RCTMergeRecursive(mergedVal, RCTJSONParse(entry[1], &jsonError))) {
+                NSDictionary *mergingValue = RCTJSONParse(entry[1], &jsonError);
+                if (!mergingValue.count || RCTMergeRecursive(mergedVal, mergingValue)) {
```

Comment:
Can you also add a test case in [`packages/default-storage/example/examples/tests.ts`](https://github.com/react-native-async-storage/async-storage/blob/main/packages/default-storage/example/examples/tests.ts)?

Reply:
yeah, turns out merging tests were giving false positives, fixed that up in next commit